cpu/qn908x/Makefile.dep

```diff
@@ -0,0 +1,16 @@
+# Periph modules in this CPU are defined whenever possible based on the vendor
```

Comment:
This is something up for discussion. Many of the headers in the vendor/ directory define useful constants and inline functions that can be used from drivers for RIOT-OS, but sometimes the intended way to use the module in the vendor SDK is different than the standard API in RIOT-OS. Take the example of WDT in this commit, the vendor's fsl_wdt.c file was not very useful (initialization of the WDT in a different way) and therefore not included here. In other cases, like the fsl_clock.c it is really useful to have the constants like `kCLOCK_Wdt` defined in fsl_clock.h and the CLOCK_GetFreq() function which does all the proper math to figure out the current frequency of a given clock source. I think that the overall goal of reuse these modules with other variants might not be possible so I'd appreciate some guidance on whether I should try to use vendor functions from the vendor SDK when possible or just access the registers directly with the headers defined in QN908XC.h.

Reply:
I ended up removing most of the vendor SDK files in the vendor/drivers directory as they are not that useful and other drivers don't match the API in RIOT-OS so a re-implementation would be needed for parts of them. Only fsl_clock and fsl_iocon (the mux) are there now.